w10h01/test/pgdp/trials/TrialOfTheDreamsTest.java

```diff
@@ -0,0 +1,59 @@
+package pgdp.trials;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static pgdp.trials.TrialOfTheDreams.lockPick;
+
+public class TrialOfTheDreamsTest {
+    // Please note: Method lockPick(Function<Byte[], Boolean> lock, List<Byte> key, int maxlen) has to be set to public
+    // for this test to work
+
+    @Test
+    @DisplayName("Basic Example")
+    void testBasicExample() {
+        Function<Byte[], Boolean> testFunction = new Function<Byte[], Boolean>() {
```

Comment:
Could you refactor this into a helper method? For example:
```suggestion
        var testFunction = createTestFunction(bytes -> (bytes.length == 2 && bytes[0] == -97 && bytes [1] == 37))
```

Reply:
@MoLaden [These are the Java Docs for Lambdas](https://docs.oracle.com/javase/tutorial/java/javaOO/lambdaexpressions.html).